utils/src/treev2/quickPickWizard/compatibility/CompatibilityRecursiveQuickPickStep.ts

```diff
@@ -0,0 +1,115 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as types from "../../../../index";
+import { getLastNode } from "../QuickPickWizardContext";
+import { CompatibilityContextValueQuickPickStep } from './CompatibilityContextValueQuickPickStep';
+import { localize } from "../../../localize";
+import { NoResourceFoundError, UserCancelledError } from "../../../errors";
+import type { ContextValueFilterQuickPickOptions } from "../ContextValueQuickPickStep";
+import { AzExtTreeItem, isAzExtTreeItem } from "../../../tree/AzExtTreeItem";
+import { isAzExtParentTreeItem } from "../../../tree/isAzExtParentTreeItem";
+import { isWrapper } from "../../../registerCommandWithTreeNodeUnwrapping";
+
+export interface CompatibilityRecursiveQuickPickOptions extends ContextValueFilterQuickPickOptions {
+    create?: types.CreateOptions;
+}
+
+/**
+ * Recursive step which is compatible which adds create picks based if the node has {@link types.CompatibleQuickPickOptions.createChild quickPickOptions.createChild} defined.
+ */
+export class CompatibilityRecursiveQuickPickStep<TContext extends types.QuickPickWizardContext> extends CompatibilityContextValueQuickPickStep<TContext, CompatibilityRecursiveQuickPickOptions> {
```

Comment:
Woof.

Why does the recursive step also have to support createChild? Shouldn't we just have a `CompatibilityCreateQuickPickStep` instead of trying to force both of these implementations into one?

Reply:
That's fair. I drew it up this way because we can't have multiple inheritance :sob: